test/functional/ui/searchhl_spec.lua

```diff
@@ -674,4 +674,20 @@ describe('search highlighting', function()
       :%g@a/b^                                 |
     ]])
   end)
+
+  it('incsearch is still visible after :redraw from timer', function()
+    helpers.exec([[
+      call setline(1, ['foo', 'bar'])
+      autocmd CmdlineChanged * call timer_start(0, {-> execute('redraw!')})
+    ]])
+    feed('/foo<CR>/bar')
+    screen:expect {
+      grid = [[
+      foo                                     |
+      {3:bar}                                     |
+      {1:~                                       }|*4
+      /bar^                                    |
+    ]],
+    }
+  end)
```

Comment:
This test gives warnings:
```
warning: Screen changes were received after the expected state. This indicates
indeterminism in the test. Try adding screen:expect(...) (or poke_eventloop())
between asynchronous (feed(), nvim_input()) and synchronous API calls.
  - Use screen:redraw_debug() to investigate; it may find relevant intermediate
    states that should be added to the test to make it more robust.
  - If the purpose of the test is to assert state after some user input sent
    with feed(), adding screen:expect() before the feed() will help to ensure
    the input is sent when Nvim is in a predictable state. This is preferable
    to poke_eventloop(), for being closer to real user interaction.
  - poke_eventloop() can trigger redraws and thus generate more indeterminism.
    Try removing poke_eventloop().
      
stack traceback:
	test/functional/ui/screen.lua:788: in function '_wait'
	test/functional/ui/screen.lua:470: in function 'expect'
	test/functional/ui/searchhl_spec.lua:684: in function <test/functional/ui/searchhl_spec.lua:678>
```
Maybe just use an RPC event instead?
```suggestion

  it('incsearch is still visible after :redraw from K_EVENT', function()
    fn.setline(1, { 'foo', 'bar' })
    feed('/foo<CR>/bar')
    screen:expect([[
      foo                                     |
      {3:bar}                                     |
      {1:~                                       }|*4
      /bar^                                    |
    ]])
    command('redraw!')
    -- There is an intermediate state where :redraw! removes 'incsearch' highlight.
    screen:expect_unchanged(true)
  end)
```

Reply:
Yeah I was still looking into it, thanks!